src/universe/GUTVisitor.java

```diff
@@ -369,7 +369,8 @@ private boolean isCompatibleCastInInfer(AnnotatedTypeMirror castType, AnnotatedT
             // Special handling for case with two ConstantSlots: even though they may not be comparable,
             // but to infer more program, let this case fall back to "anycast" silently and continue
             // inference.
-            return constraintManager.getConstraintVerifier().areComparable(castCSSlot, exprCSSlot);
+            return constraintManager.addSubtypeConstraintNoErrorMsg(castCSSlot, exprCSSlot) ||
+                constraintManager.addSubtypeConstraintNoErrorMsg(exprCSSlot, castCSSlot);
```

Comment:
Not sure if it would be better here to just add a
`addComparableConstraintNoErrorMsg` to CFI like the `isSubType` [calls](https://github.com/opprop/checker-framework-inference/blob/58af3404543ce73b88e984e1c24229d9d4c13a80/src/checkers/inference/model/ConstraintManager.java#L196)